agent/structs/config_entry_intentions.go

```diff
@@ -535,39 +535,63 @@ func (e *ServiceIntentionsConfigEntry) HasAnyPermissions() bool {
 	return false
 }
 
+// Formats errors produced by validate. Examples:
+// - ('Destination', 'Name', 'my error') => 'intention validation error: Destination.Name: my error'
+// - ('Sources[0]', "", 'lorem ipsum')   => 'intention validation error: Sources[0]: lorem ipsum'
+func wrapValidationError(position string, field string, err error) error {
+	if err == nil {
+		return nil
+	}
+
+	invalidFieldString := ""
+	if field != "" {
+		invalidFieldString = "." + field
+	}
+
+	return fmt.Errorf("intention validation error: %s%s: %v", position, invalidFieldString, err)
+}
+
+func wrapValidationErrorString(position string, field string, errString string) error {
+	return wrapValidationError(position, field, fmt.Errorf(errString))
+}
+
```

Comment:
I have some reservations about abstracting away error handling to helper funcs. IMO error handling should be kept simple (even at the cost of repetition) and close to the `if err != nil` check (as-is, a dev will have to step into 1-2 funcs to see what's being done with the args).

My gut feeling is to keep the verbose error-handling as-is (with added Destination/Source info) but it's probably best to get more opinions!

An alternative solution might be to define a custom error type so you can do something like
```
return &customErr{
    pos:  "Destination",
    field: "Name",
    msg: "service name is required and cannot be empty"
}
```
which is a bit more idiomatic and upfront about what the fields are.

Reply:
> My gut feeling is to keep the verbose error-handling as-is (with added Destination/Source info) but it's probably best to get more opinions!

It certainly wouldn't be too difficult to rewrite this PR to directly use `fmt.Errorf`, it would just have a lot of copy-paste formatting and I was concerned about the maintainability of consistent error messages.

I think you've proposed a good alternative solution -- defining a custom error type. The `Error()` method would enforce consistent formatting. What are your thoughts on defining the `customErr` struct on one-line versus multiple? (I feel a bit weird about expanding all of the existing one-line messages into 5 lines, but I'm not super familiar with Golang conventions for this kind of thing.)